stacks-common/build.rs

```diff
@@ -0,0 +1,43 @@
+use std::path::Path;
+use std::{env, fs};
+
+use toml::Value;
+
+fn main() {
+    let toml_content =
+        fs::read_to_string("../versions.toml").expect("Failed to read versions.toml");
+
+    let config: Value = toml::from_str(&toml_content).expect("Failed to parse TOML");
+
+    let mut rust_code = String::from("// Auto-generated code from versions.toml\n\n");
+
+    match config {
+        Value::Table(table) => {
+            for (key, val) in table {
+                match val {
+                    Value::String(s) => {
+                        let const_value = format!("\"{}\"", s);
+                        rust_code.push_str(&format!(
+                            "pub const {}: &str = {};\n",
+                            key.to_uppercase(),
+                            const_value
+                        ));
+                    }
+                    _ => {
+                        panic!("Invalid value type in versions.toml: {:?}", val);
+                    }
+                };
+            }
+        }
+        _ => {
+            panic!("Invalid value type in versions.toml: {:?}", config);
+        }
+    }
```

Comment:
A few things can be improved here:

- Use `let else` to reduce nesting and empty lines
- No need to call `format!` twice
- Use the `Debug` print to add double-quotes

```suggestion
let Value::Table(table) = config else {
    panic!("Invalid value type in versions.toml: {config:?}");
}

for (key, val) in table {
    let Value::String(s) = val else {
        panic!("Invalid value type in versions.toml: {val:?}");
    }

    rust_code.push_str(&format!(
        "pub const {}: &str = {s:?};\n",
        key.to_uppercase()
    )); 
}